### PR TITLE
[chore] Improve events for Server  

### PR DESCRIPTION
**Pull request by MUzairS15**
**Notes for Reviewers**

This PR fixes #
1. Persists k8s context instead of skipping whenever possible, earlier we were persisting the context if all checks passed, the behaviour is changed to persist the context if we are able to get server ID, server ID is mandated because it is one of the attribute used to determine unique contexts, otherwise user will quickly have stale connections.
2. Send the K8s deployment error field path separately as a short description instead of shoving the entire short description into one.
3. Disables Connections page from the navigator menu.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — DISMISSED** [review]

**Comment by leecalcote** [comment]
@MUzairS15 there are some failed check that need investigating...

**Comment by leecalcote** [comment]
Remove unused package. Enhanced error descriptions.

**Comment by leecalcote** [comment]
@MUzairS15 this unit test will need updated - https://github.com/meshery/meshery/actions/runs/6266860852/job/17018711297?pr=8864#step:5:259. @Philip-21 I wonder if you might be able to help move this forward.

**Comment by Philip-21** [comment]
> Member

@leecalcote unit tests fix have been implmented [here](https://github.com/meshery/meshery/pull/8585)

**Review by Philip-21 — COMMENTED** [review]
@MUzairS15  why is the connection page disabled from navigation menu

**Comment by leecalcote** [comment]
> @MUzairS15 why is the connection page disabled from navigation menu

It turns out that it’s just not ready for prime time (it has too many issues and too many things that don’t make sense), and that we are best off not showing it to users in its current state. // @senali-d @yash37158 @aabidsofi19 @theBeginner86

File: server/models/error.go [hunk 1]
```diff
@@ -353,19 +353,19 @@ func ErrShareDesign(err error) error {
...
 }
 
 func ErrUnreachableKubeAPI(err error, server string) error {
-	return errors.New(ErrUnreachableKubeAPICode, errors.Alert, []string{fmt.Sprintf("Error communicating with KubeAPI at %s.", server)}, []string{err.Error()}, []string{"The API server is not exposed.", "Credentials are malformed / invalid."}, []string{"Ensure client credential is not expired.", "Recheck the API server url.", "Remove the cluster credential and enable 'insecure-skip-tls-verify'."})
+	return errors.New(ErrUnreachableKubeAPICode, errors.Alert, []string{fmt.Sprintf("Error communicating with KubeAPI at %s.", server)}, []string{err.Error()}, []string{"The Kubernetes API server is not reachable.", "Credentials are invalid."}, []string{"Verify network connectivity and Kubernetes API responsiviness between Meshery Server and your cluster.", "Ensure client credential is not expired and is properly formed.", "Remove the cluster credential and enable 'insecure-skip-tls-verify'."})
```
Comment (MUzairS15) [inline server/models/error.go:368]:
```suggestion
	return errors.New(ErrUnreachableKubeAPICode, errors.Alert, []string{fmt.Sprintf("Error communicating with KubeAPI at %s.", server)}, []string{err.Error()}, []string{"The Kubernetes API server is not reachable.", "Credentials are invalid."}, []string{"Verify network connectivity and Kubernetes API responsiveness between Meshery Server and your cluster.", "Ensure client credential is not expired and is properly formed.", "Remove the cluster credential and enable 'insecure-skip-tls-verify'."})
```

**Comment by MUzairS15** [comment]
Fixed typo.

**Pull request merged**